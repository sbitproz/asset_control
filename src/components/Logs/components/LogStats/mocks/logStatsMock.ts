export const logsStatsMock = {
  INFO: 32100,
  WARNING: 33400,
  ERROR: 34500,
};
