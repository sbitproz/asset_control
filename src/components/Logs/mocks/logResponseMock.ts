export const logsResponseMock = {
  logs: [
    {
      dateTime: "2019-07-30 03:02:20,994",
      logType: "INFO",
      message:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
    },
    {
      dateTime: "2019-11-02 08:42:17,994",
      logType: "INFO",
      message: "Services",
    },
    {
      dateTime: "2019-11-30 10:15:23,994",
      logType: "ERROR",
      message: "Accounting",
    },
    {
      dateTime: "2019-09-03 22:24:28,994",
      logType: "INFO",
      message: "Sales",
    },
    {
      dateTime: "2020-06-08 04:23:35,994",
      logType: "INFO",
      message: "Human Resources",
    },
    {
      dateTime: "2020-05-02 13:19:02,994",
      logType: "WARNING",
      message: "Sales",
    },
    {
      dateTime: "2019-09-03 22:26:12,994",
      logType: "WARNING",
      message: "Human Resources",
    },
    {
      dateTime: "2020-05-31 06:07:54,994",
      logType: "INFO",
      message: "Accounting",
    },
    {
      dateTime: "2020-04-16 13:38:36,994",
      logType: "ERROR",
      message: "Sales",
    },
    {
      dateTime: "2020-04-12 19:04:36,994",
      logType: "INFO",
      message: "Product Management",
    },
    {
      dateTime: "2020-01-05 06:57:20,994",
      logType: "WARNING",
      message: "Product Management",
    },
    {
      dateTime: "2020-02-24 13:32:10,994",
      logType: "ERROR",
      message: "Research and Development",
    },
    {
      dateTime: "2020-05-27 12:37:20,994",
      logType: "WARNING",
      message: "Human Resources",
    },
    {
      dateTime: "2020-05-31 11:42:17,994",
      logType: "WARNING",
      message: "Research and Development",
    },
    {
      dateTime: "2019-08-17 04:15:32,994",
      logType: "INFO",
      message: "Product Management",
    },
    {
      dateTime: "2020-01-27 23:34:03,994",
      logType: "ERROR",
      message: "Services",
    },
    {
      dateTime: "2019-12-08 04:56:25,994",
      logType: "ERROR",
      message: "Human Resources",
    },
    {
      dateTime: "2020-02-05 19:03:52,994",
      logType: "ERROR",
      message: "Sales",
    },
    {
      dateTime: "2019-07-17 12:03:27,994",
      logType: "WARNING",
      message: "Research and Development",
    },
    {
      dateTime: "2019-11-18 21:28:01,994",
      logType: "WARNING",
      message: "Business Development",
    },
    {
      dateTime: "2019-09-11 22:55:38,994",
      logType: "ERROR",
      message: "Sales",
    },
    {
      dateTime: "2019-12-23 15:48:29,994",
      logType: "WARNING",
      message: "Accounting",
    },
    {
      dateTime: "2020-06-29 04:56:42,994",
      logType: "ERROR",
      message: "Business Development",
    },
    {
      dateTime: "2019-07-26 08:25:19,994",
      logType: "INFO",
      message: "Sales",
    },
    {
      dateTime: "2020-02-23 06:46:09,994",
      logType: "WARNING",
      message: "Engineering",
    },
  ],
  counter: {
    INFO: 32100,
    WARNING: 33400,
    ERROR: 34500,
  },
  pagination: {
    count: 25,
    offset: 1,
    limit: 25,
    total: 9999999,
  },
};

export const apiResponseMock = {
  data: logsResponseMock,
  status: 200,
  statusText: "COMPLETED",
  headers: undefined,
  config: {},
};
