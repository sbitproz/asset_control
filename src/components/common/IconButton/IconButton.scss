@import "../../../styles/abstracts/variables";

@keyframes fadeIn {
  0% {
    background-color: $color-white;
  }
  100% {
    background-color: $color-grey-lightest;
  }
}

$icon-size: 2rem;

.icon-button {
  width: $icon-size;
  height: $icon-size;
  cursor: pointer;
  display: flex;
  justify-content: center;
  align-items: center;

  &--disabled {
    opacity: $diabled-opacity;
    pointer-events: none;
    cursor: default;
  }

  &:hover {
    background-color: $color-grey-lightest;
    animation: fadeIn ease 0.5s;
  }
}
