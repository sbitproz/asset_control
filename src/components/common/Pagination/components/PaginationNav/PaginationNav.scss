@import "../../../../../styles/abstracts/variables";

.pagination-nav {
  width: 150px;
  display: flex;
  justify-content: space-between;
  font-size: 1.25rem;
}
