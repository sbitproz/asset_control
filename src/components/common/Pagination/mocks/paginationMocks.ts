export const pagniationInfoMocks = {
  count: 25,
  offset: 1,
  limit: 25,
  total: 9999999,
};
