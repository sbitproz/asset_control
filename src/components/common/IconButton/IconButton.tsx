import React, { memo } from "react";

// Styles
import "./IconButton.scss";

interface IconButtonProps {
  testid?: string;
  disabled?: boolean;
  onClick: () => void;
  children: React.ReactNode | React.ReactNodeArray;
  className?: string;
}

const IconButton: React.SFC<IconButtonProps> = ({
  testid = "",
  disabled,
  children,
  onClick,
  className = "",
}) => {
  return (
    <div
      data-testid={testid}
      className={`icon-button ${
        disabled ? "icon-button--disabled" : ""
      } ${className}`}
      onClick={onClick}
    >
      {children}
    </div>
  );
};

export default memo(IconButton);
