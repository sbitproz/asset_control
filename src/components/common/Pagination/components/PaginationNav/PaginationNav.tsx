import React, { memo } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faAngleRight,
  faAngleLeft,
  faAngleDoubleRight,
  faAngleDoubleLeft,
} from "@fortawesome/free-solid-svg-icons";

// Own
import { PaginationProperties } from "../../interfaces/paginationInterfaces";
import { IconButton } from "../../../IconButton";

// Styles
import "./PaginationNav.scss";

export interface PaginationInfoProps extends PaginationProperties {
  onClickFirst: () => void;
  onClickPrior: () => void;
  onClickNext: () => void;
  onClickLast: () => void;
}

const PaginationNav: React.SFC<PaginationInfoProps> = ({
  pageInformation,
  onClickFirst,
  onClickPrior,
  onClickNext,
  onClickLast,
}) => {
  const isFirstRecord = (): boolean => pageInformation.offset === 0;
  const isLastRecord = (): boolean =>
    pageInformation.offset + pageInformation.count >= pageInformation.total;

  return (
    <div data-testid="pagination-nav" className="pagination-nav">
      <IconButton
        testid="pagination-nav-first"
        disabled={isFirstRecord()}
        onClick={onClickFirst}
      >
        <FontAwesomeIcon icon={faAngleDoubleLeft} />
      </IconButton>

      <IconButton
        testid="pagination-nav-prior"
        disabled={isFirstRecord()}
        onClick={onClickPrior}
      >
        <FontAwesomeIcon icon={faAngleLeft} />
      </IconButton>

      <IconButton
        testid="pagination-nav-next"
        disabled={isLastRecord()}
        onClick={onClickNext}
      >
        <FontAwesomeIcon icon={faAngleRight} />
      </IconButton>

      <IconButton
        testid="pagination-nav-last"
        disabled={isLastRecord()}
        onClick={onClickLast}
      >
        <FontAwesomeIcon icon={faAngleDoubleRight} />
      </IconButton>
    </div>
  );
};

export default memo(PaginationNav);
