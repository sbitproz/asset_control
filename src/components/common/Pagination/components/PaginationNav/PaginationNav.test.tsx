import React from "react";
import { shallow } from "enzyme";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

// Own
import PaginationNav from "./PaginationNav";
import { pagniationInfoMocks } from "../../mocks/paginationMocks";
import { IconButton } from "../../../IconButton";
import { testSelector } from "../../../../../tests/testHelper";

describe("PaginationNav", () => {
  let wrapper: any;

  const [onFirst, onPrior, onNext, onLast] = [
    jest.fn(),
    jest.fn(),
    jest.fn(),
    jest.fn(),
  ];
  const [FIRST, PRIOR, NEXT, LAST] = [0, 1, 2, 3];

  beforeEach(() => {
    wrapper = shallow(
      <PaginationNav
        pageInformation={pagniationInfoMocks}
        onClickFirst={onFirst}
        onClickPrior={onPrior}
        onClickNext={onNext}
        onClickLast={onLast}
      />
    );
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render four buttons", () => {
    expect(wrapper.find(IconButton)).toHaveLength(4);
  });

  it("should render four buttons", () => {
    expect(wrapper.find(testSelector("pagination-nav")).exists()).toBe(true);
  });

  it("should render four buttons", () => {
    expect(wrapper.find(FontAwesomeIcon)).toHaveLength(4);
  });

  it("should call onFirst when first button is click", () => {
    wrapper.find(IconButton).at(FIRST).simulate("click");
    expect(onFirst).toHaveBeenCalled();
  });

  it("should call onPrior when prior button is click", () => {
    wrapper.find(IconButton).at(PRIOR).simulate("click");
    expect(onPrior).toHaveBeenCalled();
  });

  it("should call onNext when next button is click", () => {
    wrapper.find(IconButton).at(NEXT).simulate("click");
    expect(onNext).toHaveBeenCalled();
  });

  it("should call onLast when last button is click", () => {
    wrapper.find(IconButton).at(LAST).simulate("click");
    expect(onLast).toHaveBeenCalled();
  });

  it("should disable first and prior when offset is 0", () => {
    wrapper.setProps({
      pageInformation: { ...pagniationInfoMocks, offset: 0 },
    });
    expect(wrapper.find(IconButton).at(FIRST).props().disabled).toBe(true);
    expect(wrapper.find(IconButton).at(PRIOR).props().disabled).toBe(true);
  });

  it("should disable next and last when offset is in the last iteration", () => {
    wrapper.setProps({
      pageInformation: {
        ...pagniationInfoMocks,
        offset: pagniationInfoMocks.total - 1,
      },
    });
    expect(wrapper.find(IconButton).at(NEXT).props().disabled).toBe(true);
    expect(wrapper.find(IconButton).at(LAST).props().disabled).toBe(true);
  });
});
