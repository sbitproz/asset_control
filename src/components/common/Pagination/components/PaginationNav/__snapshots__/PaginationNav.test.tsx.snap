// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationNav should match snapshot 1`] = `
<div
  className="pagination-nav"
  data-testid="pagination-nav"
>
  <Memo(IconButton)
    disabled={false}
    onClick={[MockFunction]}
    testid="pagination-nav-first"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [],
            "f100",
            "M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z",
          ],
          "iconName": "angle-double-left",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </Memo(IconButton)>
  <Memo(IconButton)
    disabled={false}
    onClick={[MockFunction]}
    testid="pagination-nav-prior"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            256,
            512,
            Array [],
            "f104",
            "M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z",
          ],
          "iconName": "angle-left",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </Memo(IconButton)>
  <Memo(IconButton)
    disabled={false}
    onClick={[MockFunction]}
    testid="pagination-nav-next"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            256,
            512,
            Array [],
            "f105",
            "M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z",
          ],
          "iconName": "angle-right",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </Memo(IconButton)>
  <Memo(IconButton)
    disabled={false}
    onClick={[MockFunction]}
    testid="pagination-nav-last"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [],
            "f101",
            "M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z",
          ],
          "iconName": "angle-double-right",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </Memo(IconButton)>
</div>
`;
